### PR TITLE
[12.0][ADD] partner_sale_pivot: Add module

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT21479

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]
If you group by month I would fillter last 365 datos, but as the user can apply his custom fillter I approve :wink:

**Review by sergio-teruel — APPROVED** [review]

File: partner_sale_pivot/views/sale_report_views.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="sale_report_view_form" model="ir.ui.view">
+         <field name="name">sale.report.partner_sale_pivot</field>
+         <field name="model">sale.report</field>
+         <field name="arch" type="xml">
+             <pivot disable_linking="True">
+                 <field name="confirmation_date" interval="month" type="col"/>
+                 <field name="product_id" type="row"/>
+                 <field name="price_subtotal" type="measure"/>
```
Comment (legalsylvain) [inline partner_sale_pivot/views/sale_report_views.xml:22]: I think it is intereresting to have by default, quantity sold, in a second measure column. Don't you think ?

**Review by legalsylvain — APPROVED** [review]
A minor suggestion inline.
LGTM otherwise. Good work from Tecnativa team, as usual.
Thanks a lot.

**Review by chienandalu — APPROVED** [review]
Ok, but please add @carlosdauden and @legalsylvain suggestions :)

**Review by rousseldenis — APPROVED** [review]

**Review by legalsylvain — CHANGES_REQUESTED** [review]
cf. previous comment.

File: partner_sale_pivot/views/sale_report_views.xml (hunk 1 above)
Comment (ernestotejeda) [reply partner_sale_pivot/views/sale_report_views.xml:22]: What would be 'quantity sold'?
Screenshot from 2020-02-04 08-32-28

**Comment by ernestotejeda** [comment]
> If you group by month I would fillter last 365 datos, but as the user can apply his custom fillter I approve

Do you mean set a default filter for the Confirmation Date in the search view, so that the user can select the default filter 'Last 365 days'? Something like this but for the confirmation date?

Screenshot from 2020-02-04 08-45-28

File: partner_sale_pivot/views/sale_report_views.xml (hunk 1 above)
Comment (legalsylvain) [reply partner_sale_pivot/views/sale_report_views.xml:22]: Oh ! You're right.
I'm so V8.0 ! V12 is more complete. 
Well, I could say Ordered + Delivered + invoiced. but it's maybe to much. Don't no. What do you think ?

Comment (ernestotejeda) [reply partner_sale_pivot/views/sale_report_views.xml:22]: In my opinion the table become large, also users can get that with a few clicks

**Review by legalsylvain — APPROVED** [review]

**Comment by carlosdauden** [comment]
Yes, my poposal is add:
`<filter string="Confirmed in last 365 days" name="confirm_date_last_365_days" date="confirm_date" default_period="last_365_days"/>`

and replace action context witdh:
`{'search_default_partner_id': active_id, 'search_default_confirm_date_last_365_days':1}`

**Comment by carlosdauden** [comment]
Yes, my poposal is add filter with **default_period="last_365_days"** to the search view:
`<filter string="Confirmed in last 365 days" name="confirm_date_last_365_days" date="confirm_date" default_period="last_365_days"/>`

and replace action context witdh:
`{'search_default_partner_id': active_id, 'search_default_confirm_date_last_365_days':1}`

**Comment by pedrobaeza** [comment]
@ernestotejeda please finish this

**Comment by ernestotejeda** [comment]
@carlosdauden I left the filter string value only with "Confirmed in". Check it out and tell me what do you think.

**Comment by carlosdauden** [comment]
@ernestotejeda Perfect :+1:

**Pull request merged**